docs/source/en/pipeline_webserver.mdx

```diff
@@ -0,0 +1,160 @@
+    while True:
+        (string, response_q) = await q.get()
+        out = pipe(string)
+        await response_q.put(out)
+
+
+app = Starlette(
+    routes=[
+        Route("/", homepage, methods=["POST"]),
+    ],
+)
+
+
+@app.on_event("startup")
+async def startup_event():
+    q = asyncio.Queue()
+    app.model_queue = q
+    asyncio.create_task(server_loop(q))
+```
+
+Now you can start with
+```bash
+uvicorn server:app
+```
+
+And you can query it:
+```bash
+curl -X POST -d "test [MASK]" http://localhost:8000/
+#[{"score":0.7742936015129089,"token":1012,"token_str":".","sequence":"test."},...]
+```
+
+And there you go you have a first idea of how to create a webserver.
+
+What is really important is that we load the model only **once** so there is no copies
+of the model on the webserver, so no unnecessary RAM being used.
+Then the queuing mecanism allows you to do fancy stuff like maybe accumulating a few
+items before inferring to use dynamic batching:
+
+```py
+(string, rq) = await q.get()
+strings = []
+queues = []
+while True:
+    try:
+        (string, rq) = await asyncio.wait_for(q.get(), timeout=0.001)  # 1ms
+    except asyncio.exceptions.TimeoutError:
+        break
+    strings.append(string)
+    queues.append(rq)
+strings
+outs = pipe(strings, batch_size=len(strings))
+for (rq, out) in zip(queues, outs):
+    await rq.put(out)
+```
+
+<Tip warning={true}>
+Do not activate this without checking it makes sense for your load
+</Tip>
+
+The propose code is optimized for readability, not for being the best code.
+First of all there's no limit in the size of the batch, this is usually not a 
+great idea. Then the timeout is reset on every fetch of the queue, meaning you could
+wait much more than 1ms before running the inference (delaying the first request 
+by that much), it would be better to have a single 1ms deadline.
+
+This will always wait for 1ms even if the queue is empty, which might not be the
+best, you probably want to start doing the inference if there's nothing in the queue.
+But maybe it does make sense if batching is really crucial for your use case.
+Again there's no really one best solution.
+
+
+## Few things you might want to consider
+
+### Error checking
+
+There's a lot that can go wrong in production, out of memory, out of space
+loading the model might fail, the query might be wrong, the query might be
+correct but still fail to run because of a model misconfigurations etc...
+
+In general it's good if the servers does output the errors to the user so
+adding a lot of `try..except` to show those errors to your end users is a good
+idea. But also maybe not all of them depending on your security context.
+
+### Circuit breaking
+
+Webservers usually look better when they do circuit breaking. It means they 
+return proper errors while under too much load instead of just waiting for the query for
+infinity. Return 503 instead of waiting for super long or 504 after a long time.
```

Comment:
```suggestion
return proper errors when they're overloaded instead of just waiting for the query indefinitely. Return a 503 error instead of waiting for a super long time or a 504 after a long time.
```